### PR TITLE
Fix translation in GetSocketOption with SocketOptionName.Error

### PR DESCRIPTION
**Pull request by stephentoub**
On Unix we're currently handing back the underlying errno value from the platform.  We need to instead translate that to a SocketError value.

Also added a test.

Fixes https://github.com/dotnet/corefx/issues/8464
cc: @ericeil, @davidsh, @cipop, @roji

### CODE REVIEW

**Comment by stephentoub** [comment]
Test Innerloop OSX Release Build and Test please (build machine went offline)
Test Innerloop Windows_NT Debug Build and Test please (https://github.com/dotnet/corefx/issues/6388)

**Comment by stephentoub** [comment]
Test Innerloop Windows_NT Debug Build and Test please (build machine went down)

File: src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs [hunk 1]
```diff
@@ -981,6 +989,21 @@ public static unsafe SocketError GetSockOpt(SafeCloseSocket handle, SocketOption
                 optLen = 0;
                 err = Interop.Sys.GetSockOpt(handle, optionLevel, optionName, null, &optLen);
             }
+            else if (optionName == SocketOptionName.Error && optionValue.Length >= sizeof(int))
+            {
+                int outError;
+                SocketError returnError = GetSockOpt(handle, optionLevel, optionName, out outError);
```
Comment (ericeil) [inline src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs]: Why isn't this calling `GetSocketErrorOption` like you did in the other method?

**Comment by ericeil** [comment]
Looks good, other than one question.

File: src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs]: This is calling the other method, which calls GetSocketErrorOption.  I was trying to keep the logic around calling GetSocketErrorOption and doing the translation consolidated in one place.  This method then takes that result and puts the int into the byte[] array.

Comment (ericeil) [inline src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs]: Oh, that makes sense.  :)

**Pull request merged**